app/src/main/java/com/arygm/quickfix/ui/search/Announcement.kt

```diff
@@ -1,33 +1,308 @@
-import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.PaddingValues
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
 import androidx.compose.foundation.layout.padding
-import androidx.compose.material3.ExperimentalMaterial3Api
+import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.layout.width
+import androidx.compose.foundation.rememberScrollState
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.foundation.verticalScroll
+import androidx.compose.material3.Button
+import androidx.compose.material3.ButtonDefaults
+import androidx.compose.material3.Icon
+import androidx.compose.material3.MaterialTheme
 import androidx.compose.material3.MaterialTheme.colorScheme
 import androidx.compose.material3.Scaffold
 import androidx.compose.material3.Text
 import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
 import androidx.compose.ui.Alignment
 import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.draw.shadow
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.graphics.graphicsLayer
 import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.text.TextStyle
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.zIndex
+import com.arygm.quickfix.R
+import com.arygm.quickfix.ui.elements.QuickFixButton
+import com.arygm.quickfix.ui.elements.QuickFixTextFieldCustom
 import com.arygm.quickfix.ui.navigation.NavigationActions
 
-@OptIn(ExperimentalMaterial3Api::class)
 @Composable
 fun AnnouncementScreen(navigationActions: NavigationActions, isUser: Boolean = true) {
-  // Use Scaffold for the layout structure
-  Scaffold(
-      containerColor = colorScheme.background,
-      content = { padding ->
-        Column(
-            modifier = Modifier.fillMaxSize().testTag("AnnouncementContent"),
-            verticalArrangement = Arrangement.Center,
-            horizontalAlignment = Alignment.CenterHorizontally) {
+
+  var title by remember { mutableStateOf("") }
+  var category by remember { mutableStateOf("") }
+  var location by remember { mutableStateOf("") }
+  var description by remember { mutableStateOf("") }
+
+  var titleIsEmpty by remember { mutableStateOf(true) }
+  var categoryIsSelected by remember { mutableStateOf(true) } // TODO: add the different categories
+  var locationIsSelected by remember { mutableStateOf(true) } // TODO: add the implemented location
+  var descriptionIsEmpty by remember { mutableStateOf(true) }
+
+  BoxWithConstraints {
+    val widthRatio = maxWidth / 411
```

Comment:
Ask @AnotherMedo for confirmation but i think we shouldn't use BoxWithConstraints, use weights instead, it's more concise

Reply:
I will do so and fix it in the next pr including the composable for uploading pictures.